dev/Security.AccessControl/Security.AccessControl/GetSecurityDescriptorForAppContainerNames.cpp

```diff
@@ -0,0 +1,116 @@
+﻿#include "pch.h"
+
+#include <wil/resource.h>
+#include <wil/result.h>
+#include "Security.AccessControl.h"
+#include <vector>
+#include <sddl.h>
+#include <AclAPI.h>
+#include <UserEnv.h>
+
+using namespace std;
+
+namespace
+{
+    wil::unique_any_psid GetCurrentProcessSid()
+    {
+        wil::unique_any_psid result;
+        wil::unique_process_heap_ptr<TOKEN_GROUPS> tg;
+        DWORD length = 0;
+        if (!::GetTokenInformation(
+            ::GetCurrentProcessToken(),
+            TokenLogonSid,
+            nullptr,
+            0,
+            &length
+        ))
+        {
+            auto err = ::GetLastError();
+            if (err != ERROR_INSUFFICIENT_BUFFER)
+            {
+                THROW_WIN32(err);
+            }
+            tg.reset(static_cast<TOKEN_GROUPS*>(::HeapAlloc(::GetProcessHeap(), HEAP_ZERO_MEMORY, length)));
+        }
```

Comment:
https://github.com/microsoft/wil/blob/master/include/wil/token_helpers.h could acquire this as a helper! (We need to eventually invert the lookup, so you can pass the token info type you want... there are several information classes that share the same output struct.)